ibis/backends/datafusion/registry.py

```diff
@@ -0,0 +1,118 @@
+from __future__ import annotations
+
+import datafusion as df
+import pyarrow as pa
+import pyarrow.compute as pc
+
+import ibis.expr.datatypes as dt
+import ibis.expr.operations as ops
+from ibis.expr.decompile import _to_snake_case
+from ibis.formats.pyarrow import PyArrowType
+
+
+def create_udf(
+    op, udf, input_types, return_type=None, volatility="immutable", name=None
+):
+    return df.udf(
+        udf,
+        input_types=input_types,
+        return_type=PyArrowType.from_ibis(op.dtype)
+        if return_type is None
+        else return_type,
+        volatility=volatility,
+        name=_to_snake_case(op.__name__) if name is None else name,
+    )
+
+
+def extract_microsecond(array: pa.Array) -> pa.Array:
+    arr = pc.multiply(pc.millisecond(array), 1000)
+    return pc.cast(pc.add(pc.microsecond(array), arr), pa.int32())
+
+
+def epoch_seconds(array: pa.Array) -> pa.Array:
+    return pc.cast(pc.divide(pc.cast(array, pa.int64()), 1000_000), pa.int32())
+
+
+def extract_down(array: pa.Array) -> pa.Array:
+    return pc.choose(
+        pc.day_of_week(array),
+        "Monday",
+        "Tuesday",
+        "Wednesday",
+        "Thursday",
+        "Friday",
+        "Saturday",
+        "Sunday",
+    )
+
+
+def extract_second(array: pa.Array) -> pa.Array:
+    return pc.cast(pc.second(array), pa.int32())
+
+
+def extract_millisecond(array: pa.Array) -> pa.Array:
+    return pc.cast(pc.millisecond(array), pa.int32())
+
+
+udfs = {
```

Comment:
```suggestion
UDFS = {
```

to give some indication that this variable is a module global.